python/srf/_pysrf/src/egress_port.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+* SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+* SPDX-License-Identifier: Apache-2.0
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <pysrf/node.hpp>
+#include <pysrf/types.hpp>
+
+#include <srf/segment/egress_port.hpp>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/pytypes.h>
+
+namespace srf::segment {
+namespace py = pybind11;
+
+template <>
+EgressPort<pysrf::PyHolder>::EgressPort(SegmentAddress address, PortName name) :
+  m_segment_address(address),
+  m_port_name(std::move(name)),
+  m_sink(std::make_unique<pysrf::PythonNode<pysrf::PyHolder, pysrf::PyHolder>>())
+{
+    this->set_name(m_port_name);
+}
```

Comment:
I dont love this template specialization for a couple of reasons:

1. It will be very easy to let the `EgressPort<pysrf::PyHolder>` object get out of sync with the normal `EgressPort<T>` since you have to reimplement the entire class just to change a single line of code.
2. This doesnt allow you to use an internal `PythonNode` with any other type `T` besides `pysrf::PyHolder`

I think this would work better with another template on `EgressPort` that accepts the internal node type to create. Then you can have:

- `EgressPort<PythonNode, pysrf::PyHolder>`
- `EgressPort<RxNode, pysrf::PyHolder>`
- `EgressPort<PythonNode, int>`
- `EgressPort<PythonNode, std::shared_ptr<MultiMessage>>`
- `EgressPort<SnrNode, int>`

Reply:
> I dont love this template specialization for a couple of reasons:
> 
>     1. It will be very easy to let the `EgressPort<pysrf::PyHolder>` object get out of sync with the normal `EgressPort<T>` since you have to reimplement the entire class just to change a single line of code.
> 
>     2. This doesnt allow you to use an internal `PythonNode` with any other type `T` besides `pysrf::PyHolder`
> 
> 
> I think this would work better with another template on `EgressPort` that accepts the internal node type to create. Then you can have:
> 
>     * `EgressPort<PythonNode, pysrf::PyHolder>`
> 
>     * `EgressPort<RxNode, pysrf::PyHolder>`
> 
>     * `EgressPort<PythonNode, int>`
> 
>     * `EgressPort<PythonNode, std::shared_ptr<MultiMessage>>`
> 
>     * `EgressPort<SnrNode, int>`

Probably worth discussing this in more detail; its not clear to me how this kind of flexibility could work from a Python binding perspective, since all possible combinations would need to be seen at compile time.